homeassistant/components/schlage/lock.py

```diff
@@ -0,0 +1,80 @@
+"""Platform for Schlage lock integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyschlage.lock import Lock
+
+from homeassistant.components.lock import LockEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import SchlageDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Schlage WiFi locks based on a config entry."""
+    coordinator: SchlageDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        [
+            SchlageLock(coordinator=coordinator, device_id=device_id)
+            for device_id in coordinator.data.locks
+        ]
```

Comment:
```suggestion
            SchlageLock(coordinator=coordinator, device_id=device_id)
            for device_id in coordinator.data.locks
```

You can use a generator expression here.

`black` will complain about this suggestion

Reply:
I think the list comprehension is fine--I'd imagine most folks have a small number of locks, so the overhead of the list is fairly low.

Reply:
I’m not sure what the objection to using a generator expression is given it’s less code?

Reply:
Done